Baseline formatter upon Quarkus format

rnc opened the pull request:
This updates the formatter (and renames them) to be based upon the Quarkus ide-config from https://github.com/quarkusio/quarkus/blob/main/independent-projects/ide-config/src/main/resources/eclipse-format.xml but then reapplies the PNC specific changes from the PRs. Combined with prior PRs, it makes a single location that is accessible by plugins to format code without having to copy the Eclipse formatter xml everywhere.

rnc committed:
Fix https prefix and README cleanup

rnc committed:
Remove cleanup profiles. Rename templates

rnc committed:
Update formatter to be baselined to https://github.com/quarkusio/quarkus/blob/main/independent-projects/ide-config/src/main/resources/eclipse-format.xml

rnc committed:
Reapply #1 align parameters one per line if too long

rnc committed:
Reapply #2 enable fluent API alignment

rnc committed:
Reapply #3 align parameters in an annotation, one per line

rnc committed:
Reapply #4 adjusted eclipse settings based on work with Matej

rnc committed:
Reapply #5 fix formatter to use 120 lines for comments

rnc committed:
Reapply #6 properly format enums

rnc committed:
Reapply #8 make annotation on enums be in the same line

rnc committed:
Reapply #9 wrap array initializer per line

rnc committed:
Update README

thescouser89 commented:
I'm not sure I'll be the best person to review this one. We'll have to loop in the entire PNC team to comment on the Quarkus Java format and if we're ok with it!

thescouser89 commented:
![image](https://github.com/user-attachments/assets/ceeb3fc9-868a-4ef1-ad4e-38ec89c495f9)
![image](https://github.com/user-attachments/assets/ae4e1c33-2ac3-4da1-9726-48a5d86f96cc)
![image](https://github.com/user-attachments/assets/339ac5bb-b1f5-4513-a367-ef14063a103e)

Looking at the diffs the new java-formatter.xml introduces, I see that the arguments of a method should now be aligned to the opening bracket '(' of the method.

Apart from that, there are no other changes.

I personally prefer the PNC way but that's a personal preference. I do like that it seems the Quarkus java-formatter.xml is using the latest format, even though I'm not sure what changed between format versions haa! :)

rnc committed:
Update profile name and fix missing method alignment (from #4)

rnc commented:
> 
> Looking at the diffs the new java-formatter.xml introduces, I see that the arguments of a method should now be aligned to the opening bracket '(' of the method.
> 
> Apart from that, there are no other changes.
> 
> I personally prefer the PNC way but that's a personal preference. I do like that it seems the Quarkus java-formatter.xml is using the latest format, even though I'm not sure what changed between format versions haa! :)

I am slightly surprised that there are no other changes to be honest! I've gone back over the file and found a missing alignment in `org.eclipse.jdt.core.formatter.alignment_for_arguments_in_method_invocation` which _might_ resolve your issue. I've also updated the profile name to `PNC`.

I've also updated the README with matching examples for spotless and formatter-maven-plugin.

@thescouser89 Can you loop in appropriate people please?

rnc committed:
Update README with spotless and formatter-maven-plugin examples

thescouser89 commented:
@michalovjan @janinko @pkocandr @vibe13 Do you have any opinions on the new format? I think overall it's the same as what we have currently, but updated to the latest eclipse codestyle version.

thescouser89 commented:
Ran it through several PNC projects and everything looks good! I'll just merge it!
